### ISSUE TITLE
Support `None` in mint authority constraints

### ISSUE THREAD

**Issue by Henry-E** [coral-xyz/anchor#3066]
`mint:freeze_authority` and `mint:authority` should be able to support None. Maybe they do already? But it's getting to be more common that people want to check that a mint has these option set to None.

**Comment by acheroncrypto** [comment]
Yeah, the `mint` constraint parser expects the expression to be an account name, so `None` support would have to be implemented separately. I support this feature.

**Comment by illegalcall** [comment]
Hey @acheroncrypto , is this issue still relevant? Can i pick this?
